### PR TITLE
Fix itemless topic tests, and relocated QoS.xml

### PR DESCRIPTION
**Pull request by dmills-lsst**

### CODE REVIEW

**Review by pkubanek — APPROVED** [review]
In high demand (as I am seeing the consequences as well), thanks for the patch!

File: lsstsal/scripts/gensimplesample.tcl [hunk 1]
```diff
@@ -523,11 +523,11 @@ global OPTIONS
      puts $fout "	struct ackcmd \{"
      add_private_idl $fout
      puts $fout "	  long	ack;
-	  long	error;
+	  long		error;
```
Comment (pkubanek) [inline lsstsal/scripts/gensimplesample.tcl:526]: Can you please revert formatting changes?

File: lsstsal/scripts/gensimplesample.tcl [hunk 2]
```diff
@@ -548,12 +548,12 @@ global SAL_WORK_DIR OPTIONS
...
      puts $fout "	  long	salIndex;
-	  long	ack;
-	  long	error;
+	  long		ack;
+	  long		error;
 	  string	result;
 	  string	identity;
-	  long	origin;
-	  long	cmdtype;
+	  long		origin;
```
Comment (pkubanek) [inline lsstsal/scripts/gensimplesample.tcl:555]: See above, those doesn't seem as necessary

**Review by rbovill — APPROVED** [review]
LGtM

File: lsstsal/scripts/gensimplesample.tcl (hunk 1 above)
Comment (rbovill) [reply lsstsal/scripts/gensimplesample.tcl:526]: These formatting changes look fine in the context of the whole file.  I'd say leave them.

File: lsstsal/scripts/gensimplesample.tcl (hunk 2 above)
Comment (rbovill) [reply lsstsal/scripts/gensimplesample.tcl:555]: These formatting changes look fine in the context of the whole file. I'd say leave them.

**Pull request merged**